opam installer installs too-old version of ocaml

I followed the instructions at http://opam.ocaml.org/doc/Install.html to install opam, i.e.

    wget https://raw.github.com/ocaml/opam/master/shell/opam_installer.sh -O - | sh -s /usr/local/bin

but this installs ocaml 4.02.1 which is slightly too old, and fails to compile F*.

I actually have a suitable version of ocaml installed via my system's package manager (I'm on Fedora Linux) but it doesn't get used because opam puts its own directory first on the `PATH`.

I tried reordering the `PATH` to put the opam directory last, but then I get this error instead:

```
File "lib/ml/FStar_Char.ml", line 1:
Error: The files /usr/lib64/ocaml/pervasives.cmi
       and /home/robin/.opam/4.02.1/lib/batteries/batChar.cmi
       make inconsistent assumptions over interface CamlinternalFormatBasics
Command exited with code 2.
Compilation unsuccessful after building 22 targets (6 cached) in 00:00:00.
Makefile:93: recipe for target 'fstar.exe' failed
```

--- a/INSTALL.md
+++ b/INSTALL.md
@@ -268,7 +268,14 @@ that's over there (it's optimized for F*).
    - If you're on Windows see https://github.com/protz/ocaml-installer/wiki
      for instructions on how to configure your environment for use with OPAM
 
-3. F* depends on a bunch of external OCaml packages which you can install using OPAM:
+3. Ensure that OPAM is using a recent enough version of OCaml
+
+   - Type `opam switch list`. The current OCaml version used by opam
+     is identified by the letter C. If it is not within the version
+     range required by F* (see above), type `opam switch ` and then
+     the version number you wish to switch opam to.
+
+4. F* depends on a bunch of external OCaml packages which you can install using OPAM:
 
   ```sh
   $ opam install ocamlfind batteries stdint zarith yojson fileutils pprint

